.github/workflows/main.yml

```diff
@@ -330,7 +334,9 @@ jobs:
       # MacOS doesn't support x86 from Xcode 10 onwards.
       - name: Build & Test x86
         if: (runner.os == 'Linux' && matrix.container == '') || (runner.os == 'Windows' && matrix.compiler != 'clang-cl')
-        run: python3 .github/workflows/build_and_test.py ${{ runner.os }} x86 ${{ matrix.compiler }} ${{ matrix.version }}
+        run:
+          python3 -m pip install packaging==25.0
```

Comment:
IMHO it is cleaner to separate the installation of build dependencies and test execution, makes it easier to spot what failed when the pipeline breaks.